src/engine/Service.cpp

```diff
@@ -144,15 +147,19 @@ Result Service::computeResult([[maybe_unused]] bool requestLaziness) {
     resVariables = jsonResult["head"]["vars"].get<std::vector<std::string>>();
     resBindings =
         jsonResult["results"]["bindings"].get<std::vector<nlohmann::json>>();
-  } catch (const nlohmann::json::parse_error& e) {
+  } catch (nlohmann::json::parse_error&) {
     throw std::runtime_error(absl::StrCat(
         "Failed to parse the Service result as JSON. First 100 bytes: ",
-        jsonStr.substr(0, std::min(100, static_cast<int>(jsonStr.size())))));
-  } catch (const nlohmann::json::type_error& e) {
-    throw std::runtime_error(
-        "JSON result does not have the expected structure");
+        std::string(jsonStr.data(),
+                    std::min(100, static_cast<int>(jsonStr.size())))));
```

Comment:
```suggestion
        std::string_view{jsonStr.data()}.substr(0, 100);
```
Cheaper (no string copy) and simpler (substr automatically checks for the size).
You can even compute this string_view only once, for both cases.
Or implement a lambda that does the throwing for you and just takes the initial string_view that is different for the two cases
(The throw, runtime error, StrCat, substr are always the same for both paths).

(I am very much against code duplication as it used to be a root of very bad evil in this codebase:)